x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/predicate/operator/comparison/InMapper.java

```diff
@@ -106,7 +106,12 @@ private static Block evalWithNulls(boolean[] values, BitSet nulls, boolean nullI
                         nulls.set(i);
                     } // else: leave nulls as is
                 }
-                return new BooleanArrayBlock(values, values.length, null, nulls, Block.MvOrdering.UNORDERED);
+                if (nulls.isEmpty()) {
+                    // see https://github.com/elastic/elasticsearch/issues/99347
+                    return new BooleanArrayVector(values, values.length).asBlock();
```

Comment:
I think it's worth adding a test case that reproduces this one every time. And probably explaining why we want this statement rather than a link to the bug. It takes a bit of digging to reverse the sequence of events from the bug but it's fast to read a one liner like "no nulls, it's all true." Actually, do we only get here if everything is `true`? Maybe we should just return a constant block in that case?